pkg/storage/etcd/etcd_helper.go

```diff
@@ -711,3 +710,21 @@ func (h *etcdHelper) addToCache(index uint64, obj runtime.Object) {
 		metrics.ObserveNewEntry()
 	}
 }
+
+func convertToStorageErrorIfPossible(err error, key string, rv uint64) error {
```

Comment:
"storage/" handles only four errors in special. See [here](https://github.com/kubernetes/kubernetes/blob/a428246960b83eb568bef1056dcc30d8ca0254bd/pkg/storage/errors.go#L23-L45).

Not sure about other error types ATM.